packages/sources/s3-csv-reader/src/endpoint/csv.ts

```diff
@@ -0,0 +1,67 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { config } from '../config'
+import overrides from '../config/overrides.json'
+import { transport } from '../transport/csv'
+
+export const inputParameters = new InputParameters(
+  {
+    bucket: {
+      required: true,
+      type: 'string',
+      description: 'The S3 bucket to query',
+    },
+    keyPrefix: {
+      aliases: ['pathPrefix'],
+      required: true,
+      type: 'string',
+      description:
+        'The path prefix of the file stored in S3. <Date.csv> is appended to search for older files.',
```

Comment:
will try to figure out how to make it clearer, but the idea is 
- provide `<bucket>` eg: data-feeds-s3-bucket
- provide `<path_prefix>` eg: path/to/file excluding -01-01-2024.csv
- `-01-01-2024.csv` (current date) gets appended to the path_prefix

so the resulting filename is `path/to/file-01-01-2024.csv` for example

I also thought about adding an `APPEND_DATE_CSV` env var defaulting to true in case we wanted to reuse this without dates, what do you think?

Reply:
I recommend specifying that we expect the file name to have the format `<name>-dd-MM-yyyy.csv`. Could we simplify this to say that we just pass in a file path to a folder, and each file has to be named just the `dd-MM-yyyy.csv`?

Reply:
yeah let's do this in a follow up, there will be a change on the infra side to make as well.